From 80766c948080a382bb752f0cb90623073626aa05 Mon Sep 17 00:00:00 2001
From: Gergely Nagy <algernon@madhouse-project.org>
Date: Wed, 23 Sep 2015 10:12:43 +0200
Subject: [PATCH] Dh_Lib: Export _DH_DO_PACKAGES before running executable
 scripts

In order to let executable scripts discover what package is being built,
so they can avoid side-effects in case a package is skipped, export
_DH_DO_PACKAGES before running them.

This, along with changes to dh-exec, will close #698054.

Signed-off-by: Gergely Nagy <algernon@madhouse-project.org>
---
 Debian/Debhelper/Dh_Lib.pm | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Debian/Debhelper/Dh_Lib.pm b/Debian/Debhelper/Dh_Lib.pm
index a5743e9..03553ce 100644
--- a/Debian/Debhelper/Dh_Lib.pm
+++ b/Debian/Debhelper/Dh_Lib.pm
@@ -760,7 +760,9 @@ sub filedoublearray {
 	if ($x) {
 		require Cwd;
 		my $cmd=Cwd::abs_path($file);
+		$ENV{"_DH_DO_PACKAGES"} = join(",", @{$dh{"DOPACKAGES"}});
 		open (DH_FARRAY_IN, "$cmd |") || error("cannot run $file: $!");
+		delete $ENV{"_DH_DO_PACKAGES"};
 	}
 	else {
 		open (DH_FARRAY_IN, $file) || error("cannot read $file: $!");
-- 
2.5.1

